pypeit/core/flux_calib.py

```diff
@@ -761,12 +752,49 @@ def fit_zeropoint(wave, counts, counts_ivar, counts_mask, exptime, airmass, std_
             Exposure time in seconds
         airmass (float):
             Airmass
-        std_dict (dict):
-            Dictionary containing information about the standard star returned by flux_calib.get_standard_spectrum
         longitude (float):
             Telescope longitude, used for extinction correction.
         latitude (float):
             Telescope latitude, used for extinction correction
+
+    Returns:
+        Nlam_star (`numpy.ndarray`_): counts/second/Angstrom
+        Nlam_ivar_star (`numpy.ndarray`_): inverse variance of Nlam_star
+        gpm_star (`numpy.ndarray`_): good pixel mask for Nlam_star
+
+    """
+    # Create copy of the arrays to avoid modification and convert to
+    # Nlam = electrons/s/Angstrom
+    delta_wave = wvutils.get_delta_wave(wave, (wave > 1.0))
+    Nlam_star = counts/exptime/delta_wave
+    Nlam_ivar_star = delta_wave**2*counts_ivar*exptime**2
+
+    # Extinction correction
+    msgs.info("Applying extinction correction")
+    extinct = load_extinction_data(longitude,latitude)
+    ext_corr = extinction_correction(wave * units.AA, airmass, extinct)
+    # Correct for extinction
+    Nlam_star = Nlam_star * ext_corr
+    Nlam_ivar_star = Nlam_ivar_star / ext_corr ** 2
+    gpm_star = counts_mask
+    return Nlam_star, Nlam_ivar_star, gpm_star
+
+
+def fit_zeropoint(wave, Nlam_star, Nlam_ivar_star, gpm_star, std_dict,
+                  mask_abs_lines=True, polyorder=4, balm_mask_wid=10.0, nresln=20., resolution=3000.,
+                  trans_thresh=0.9, polycorrect=True, polyfunc=False, debug=False):
+
+    """
+    Function to generate the sensitivity function. This function fits
+    a bspline to the 2.5*log10(flux_std/flux_counts). The break
+    points spacing, which determines the scale of variation of the
+    sensitivity function is determined by the nresln parameter.
+
+    Args:
```

Comment:
Please document the arrays Nlam_star, Nlam_ivar_star, and gpm_star

Reply:
Done